inch.go

```diff
@@ -0,0 +1,663 @@
+package inch
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math"
+	"net/http"
+	"os"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdata/influxdb/client/v2"
+	"github.com/influxdata/influxdb/models"
+)
+
+// ErrConnectionRefused indicates that the connection to the remote server was refused.
+var ErrConnectionRefused = errors.New("connection refused")
+
+// ErrorList is a simple error aggrigator to return multiple errors as one.
+type ErrorList []error
+
+func (el ErrorList) Error() string {
+	msg := ""
+
+	for _, err := range el {
+		msg = fmt.Sprintf("%s%s\n", msg, err)
+	}
+
+	return msg
+}
+
+// Main represents the main program execution.
+type Inch struct {
+	mu             sync.Mutex
+	writtenN       int
+	startTime      time.Time
+	now            time.Time
+	timePerSeries  int64 // How much the client is backing off due to unacceptible response times.
+	currentDelay   time.Duration
+	wmaLatency     float64
+	latencyHistory []time.Duration
+	totalLatency   time.Duration
+	currentErrors  int   // The current number of errors since last reporting.
+	totalErrors    int64 // The total number of errors encountered.
+
+	Stdout io.Writer
+	Stderr io.Writer
+
+	// Client to be used to report statistics to an Influx instance.
+	clt client.Client
+
+	// Decay factor used when weighting average latency returned by server.
+	alpha float64
+
+	Verbose    bool
+	ReportHost string
+	ReportTags map[string]string
+	DryRun     bool
+	MaxErrors  int
+
+	Host             string
+	Consistency      string
+	Concurrency      int
+	Measurements     int   // Number of measurements
+	Tags             []int // tag cardinalities
+	PointsPerSeries  int
+	FieldsPerPoint   int
+	BatchSize        int
+	TargetMaxLatency time.Duration
+
+	Database      string
+	ShardDuration string        // Set a custom shard duration.
+	TimeSpan      time.Duration // The length of time to span writes over.
+	Delay         time.Duration // A delay inserted in between writes.
+}
```

Comment:
Can you name this something more representative? Maybe `Simulator` or `Process` or something.